control-plane/agents/src/bin/core/volume/scheduling.rs

```diff
@@ -14,12 +14,40 @@ use crate::{
     nexus::scheduling::target_node_candidates,
     node::wrapper::NodeWrapper,
 };
+
 use agents::errors::{NotEnough, SvcError};
 use stor_port::types::v0::{
     store::{nexus::NexusSpec, volume::VolumeSpec},
-    transport::{NodeId, Replica, VolumeState},
+    transport::{NodeId, NodeTopology, PoolTopology, Replica, VolumeState},
 };
 
+use std::collections::HashMap;
+
+/// Return a list of pre sorted pools to be used by a volume at time of creation.
+pub(crate) async fn volume_pool_candidates_new(
+    request: impl Into<GetSuitablePools>,
+    registry: &Registry,
+) -> Vec<Vec<PoolWrapper>> {
+    let mut result: Vec<Vec<PoolWrapper>> = Vec::new();
+    let request = request.into();
+    let is_affinity = is_affinity_requested(request.clone());
+
+    let vec_of_pool_wrappers: Vec<PoolWrapper> =
+        volume::AddVolumeReplica::builder_with_defaults(request.clone(), registry)
+            .await
+            .collect()
+            .into_iter()
+            .map(|e| e.collect())
+            .collect();
+
+    if is_affinity {
+        result = get_filtered_pools(vec_of_pool_wrappers, request, registry);
```

Comment:
Something like this..

```suggestion
        result = group_pools_by_affinity(vec_of_pool_wrappers, request, registry);
```

Reply:
done